[BUG] 3.1.0b1 regression on PyPy3 / rapidfuzz: `error: ‘__Pyx_Owned_Py_None’ was not declared in this scope`

### Describe the bug

When trying to build [rapidfuzz](https://github.com/rapidfuzz/RapidFuzz/) with Cython 3.1.0b1 on PyPy3.10 & PyPy3.11 7.3.19, I'm seeing the following build failures:

```
FAILED: src/rapidfuzz/distance/CMakeFiles/_initialize_cpp.dir/_initialize_cpp.cxx.o 
/usr/bin/g++  -pthread -D_initialize_cpp_EXPORTS -I/tmp/RapidFuzz/src/rapidfuzz -I/tmp/RapidFuzz/src/rapidfuzz/distance -isystem /usr/include/pypy3.11 -O3 -DNDEBUG -flto=auto -fno-fat-lto-objects -fPIC -Wall -Wextra -pedantic -Wno-psabi -MD -MT src/rapidfuzz/distance/CMakeFiles/_initialize_cpp.dir/_initialize_cpp.cxx.o -MF src/rapidfuzz/distance/CMakeFiles/_initialize_cpp.dir/_initialize_cpp.cxx.o.d -o src/rapidfuzz/distance/CMakeFiles/_initialize_cpp.dir/_initialize_cpp.cxx.o -c /tmp/tmpuy4r2kwd/build/src/rapidfuzz/distance/_initialize_cpp.cxx
/tmp/tmpuy4r2kwd/build/src/rapidfuzz/distance/_initialize_cpp.cxx: In function ‘int __pyx_pf_9rapidfuzz_8distance_15_initialize_cpp_13MatchingBlock___cinit__(__pyx_obj_9rapidfuzz_8distance_15_initialize_cpp_MatchingBlock*, size_t, size_t, size_t)’:
/tmp/tmpuy4r2kwd/build/src/rapidfuzz/distance/_initialize_cpp.cxx:10713:36: error: ‘__Pyx_Owned_Py_None’ was not declared in this scope
10713 |   __Pyx_TraceReturnCValue(__pyx_r, __Pyx_Owned_Py_None, 0, 0, __PYX_ERR(0, 189, __pyx_L1_error));
      |                                    ^~~~~~~~~~~~~~~~~~~
/tmp/tmpuy4r2kwd/build/src/rapidfuzz/distance/_initialize_cpp.cxx:2436:31: note: in definition of macro ‘__Pyx_TraceReturnCValue’
 2436 |       if ((1)); else { (void) convert_function; goto_error }
      |                               ^~~~~~~~~~~~~~~~
/tmp/tmpuy4r2kwd/build/src/rapidfuzz/distance/_initialize_cpp.cxx: In function ‘int __pyx_pf_9rapidfuzz_8distance_15_initialize_cpp_13MatchingBlock_1a_2__set__(__pyx_obj_9rapidfuzz_8distance_15_initialize_cpp_MatchingBlock*, PyObject*)’:
/tmp/tmpuy4r2kwd/build/src/rapidfuzz/distance/_initialize_cpp.cxx:11637:36: error: ‘__Pyx_Owned_Py_None’ was not declared in this scope
11637 |   __Pyx_TraceReturnCValue(__pyx_r, __Pyx_Owned_Py_None, 0, 0, __PYX_ERR(0, 185, __pyx_L1_error));
      |                                    ^~~~~~~~~~~~~~~~~~~
/tmp/tmpuy4r2kwd/build/src/rapidfuzz/distance/_initialize_cpp.cxx:2436:31: note: in definition of macro ‘__Pyx_TraceReturnCValue’
 2436 |       if ((1)); else { (void) convert_function; goto_error }
      |                               ^~~~~~~~~~~~~~~~
```

(and so on)

The same issue happens with both PyPy3.10 and PyPy3.11. It doesn't seem to happen with CPython. PyPy + Cython 3.0.x also seem to work fine.

### Code to reproduce the behaviour:

```
git clone --recurse-submodules https://github.com/rapidfuzz/RapidFuzz
cd RapidFuzz
pypy3.11 -m venv .venv
pip install --pre cython scikit-build-core build
python -m build -wnx
```

(note that upstream has proactive restriction to `<3.1.0`, but this seems like a legitimate problem specific to PyPy)

### Expected behaviour

_No response_

### OS

Gentoo Linux amd64

### Python version

PyPy3.10/3.11 7.3.19

### Cython version

3.1.0b1

### Additional context

_No response_

--- a/Cython/Utility/TypeConversion.c
+++ b/Cython/Utility/TypeConversion.c
@@ -103,7 +103,7 @@ static CYTHON_INLINE PyObject *__Pyx_XNewRef(PyObject *obj) {
 #endif
 }
 
-static CYTHON_INLINE PyObject *__Pyx_Owned_Py_None(b);
+static CYTHON_INLINE PyObject *__Pyx_Owned_Py_None(int b);
 static CYTHON_INLINE PyObject * __Pyx_PyBool_FromLong(long b);
 static CYTHON_INLINE int __Pyx_PyObject_IsTrue(PyObject*);
 static CYTHON_INLINE int __Pyx_PyObject_IsTrueAndDecref(PyObject*);
@@ -420,7 +420,7 @@ static CYTHON_INLINE Py_hash_t __Pyx_PyIndex_AsHash_t(PyObject* o) {
   }
 }
 
-static CYTHON_INLINE PyObject *__Pyx_Owned_Py_None(b) {
+static CYTHON_INLINE PyObject *__Pyx_Owned_Py_None(int b) {
     CYTHON_UNUSED_VAR(b);
     return __Pyx_NewRef(Py_None);
 }

--- a/Cython/Utility/TypeConversion.c
+++ b/Cython/Utility/TypeConversion.c
@@ -421,6 +421,7 @@ static CYTHON_INLINE Py_hash_t __Pyx_PyIndex_AsHash_t(PyObject* o) {
 }
 
 static CYTHON_INLINE PyObject *__Pyx_Owned_Py_None(b) {
+    CYTHON_UNUSED_VAR(b);
     return __Pyx_NewRef(Py_None);
 }
 

--- a/Cython/Utility/TypeConversion.c
+++ b/Cython/Utility/TypeConversion.c
@@ -103,7 +103,7 @@ static CYTHON_INLINE PyObject *__Pyx_XNewRef(PyObject *obj) {
 #endif
 }
 
-#define __Pyx_Owned_Py_None(b) __Pyx_NewRef(Py_None)
+static CYTHON_INLINE PyObject *__Pyx_Owned_Py_None(b);
 static CYTHON_INLINE PyObject * __Pyx_PyBool_FromLong(long b);
 static CYTHON_INLINE int __Pyx_PyObject_IsTrue(PyObject*);
 static CYTHON_INLINE int __Pyx_PyObject_IsTrueAndDecref(PyObject*);
@@ -420,6 +420,9 @@ static CYTHON_INLINE Py_hash_t __Pyx_PyIndex_AsHash_t(PyObject* o) {
   }
 }
 
+static CYTHON_INLINE PyObject *__Pyx_Owned_Py_None(b) {
+    return __Pyx_NewRef(Py_None);
+}
 
 static CYTHON_INLINE PyObject * __Pyx_PyBool_FromLong(long b) {
   return b ? __Pyx_NewRef(Py_True) : __Pyx_NewRef(Py_False);

